lib/easybill-rest-client/generic_api.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+module EasybillRestClient
+  module GenericApi
+    def self.new(resource_name, resource_class)
```

Comment:
Why make this so complicated? Wouldn't simple inheritance and two class methods do the same trick without having that awkward anonymous classes? Especially if you then later inherit from the generated classes.